Task4_MongoDB.txt

```diff
@@ -0,0 +1,346 @@
+       Task4. Indexing Restaurants Collection
+
+
+
+4.1 Create an index which will be used by this query and provide proof (from explain() or Compass UI) that the index is indeed used by the winning plan: db.restaurants.find({ name: "Glorious Food" }) 
+
+
+Query:   db.restaurants.createIndex({ name: 1 })
+Result:	 
+{
+        "createdCollectionAutomatically" : true,
+        "numIndexesBefore" : 1,
+        "numIndexesAfter" : 2,
+        "ok" : 1
+}
+
+---------------------------------------------------------------------------
+4.2 Drop index from task 4.1
+
+
+Query:   db.restaurants.createIndex({ name: 1 })
+Result:  { "nIndexesWas" : 2, "ok" : 1 }
+
+
+---------------------------------------------------------------------------
+4.3 Create an index to make this query covered and provide proof (from explain() or Compass UI) that it is indeed covered: 
+    db.restaurants.find({ restaurant_id: "41098650" }, { _id: 0, borough: 1 })

```

Comment:
You should drop index in `4.2`, than in `4.3` create new index and than perform `find` operation.

Reply:
Done